skoolkit/basic.py

```diff
@@ -114,14 +115,92 @@
     255: 'COPY'
 }
 
+def get_number(snapshot, i):
+    if snapshot[i] == 0:
+        # Small integer (signed)
+        num = _get_integer(snapshot, i)
+    else:
+        # Floating point number (signed)
+        num = _get_float(snapshot, i)
+    return num
+
+def _get_integer(snapshot, i):
```

Comment:
I disagree. I wrote this particular function to mirror the (buggy) way in which the Spectrum handles integer values. While unlikely, it is not impossible that there exists Spectrum software that has been written (intentionally or unintentionally) to rely on this bug to function "correctly". Therefore, outputting integers exactly as the Spectrum sees them is going to be more useful in figuring out how such software works than outputting integers that are technically correct, but not necessarily the same as that software expects.